lib/openstudio/workflow/util/energyplus.rb

```diff
@@ -320,20 +320,33 @@ def self.add_energyplus_output_request(workspace, idf_object)
           end
 
           allowed_unique_objects = []
-          # allowed_unique_objects << "Output:EnergyManagementSystem" # TODO: have to merge
-          # allowed_unique_objects << "OutputControl:SurfaceColorScheme" # TODO: have to merge
-          allowed_unique_objects << 'Output:Table:SummaryReports' # TODO: have to merge
-          # OutputControl:Table:Style # not allowed
-          # OutputControl:ReportingTolerances # not allowed
-          # Output:SQLite # not allowed
-
-          if allowed_unique_objects.include?(idf_object.iddObject.name) && (idf_object.iddObject.name == 'Output:Table:SummaryReports')
-            summary_reports = workspace.getObjectsByType(idf_object.iddObject.type)
-            if summary_reports.empty?
+          allowed_unique_objects << 'Output:Constructions' # replace existing
+          allowed_unique_objects << 'OutputControl:Files' # replace existing
+          allowed_unique_objects << 'OutputControl:ReportingTolerances' # replace existing
+          # allowed_unique_objects << 'OutputControl:SurfaceColorScheme' # not wrapped
+          allowed_unique_objects << 'OutputControl:Table:Style' # replace existing
+          allowed_unique_objects << 'OutputControl:Timestamp' # replace existing
+          allowed_unique_objects << 'Output:DebuggingData' # replace existing
+          allowed_unique_objects << 'Output:Diagnostics' # replace existing
+          allowed_unique_objects << 'Output:EnergyManagementSystem' # replace existing
+          allowed_unique_objects << 'Output:EnvironmentalImpactFactors' # replace existing
+          allowed_unique_objects << 'Output:JSON' # replace existing
+          allowed_unique_objects << 'Output:Schedules' # replace existing
+          allowed_unique_objects << 'Output:SQLite' # replace existing
```

Comment:
I find overriding Output:SQLite really strange. Output:SQLite was explicitly disabled in the past and it makes sense to me. I guess there's one use case where the user explicitly set the unit conversion for tabular data and you want to override that?

Output:JSON is kinda in the same bucket...

Anyways, we don't want two measures to end up clashing with each other... we may want to create "merge/check" functions for these objects.

For Output:JSON:
```
If existing is found:
    for each report_flag in [JSON, CBOR, MessagePack]:
         existing.report_flag ||= new.report_flag
else:
   add(existing)
```

For Output:SQLite: probably want to make sure that the resulting 'Option Type' is the "MAX" ( where Simple < SimpleAndTabular), and the Unit Conversion we should probably yell very loudly if existing and new don't match.

Reply:
For OutputControl:Files this is the same actually.

For Output:DebuggingData we probably to do a flag ||= new.flag as well.


Anyways, I think we need merge functions if we want to cover breaking cases.

Reply:
I understand that point of view but I guess I don't agree. Suppose I define an object in the `energyPlusOutputRequests` method to be added like so:

```
Output:JSON,
  TimeSeriesAndTabular,                   !- Option Type
    ,                                     !- Output JSON
  No,                                     !- Output CBOR
  Yes;                                    !- Output MessagePack

```

The expected behavior is that I would get the E+ default for JSON, not get CBOR, and get MessagePack. The fact that the model already had an object with certain values set is irrelevant. 

That said, I don't really care that strongly but don't want to get sucked into spending too much time on this. If you don't like this approach, I'll probably just pare it back to the single object (`OutputControl:Timestamp`) that I care about right now. I was just adding the other objects to be comprehensive and give users complete flexibility. I personally dislike that OpenStudio is so controlling and that every object needs to be specifically called out and handled in the workflow.

Reply:
I find it particularly confusing to use the ||= operator. If CBOR was previously set to true and then you try to set it to false in your reporting measure, it will remain true... how is that a good or expected outcome?

```
irb(main):006:0> foo = true
irb(main):007:0> foo ||= false
irb(main):008:0> foo
=> true
```

Reply:
Not 100% sure I follow why you disagree.

What we check for is whether the *workspace*  has this object already. That could mean it was instantiated at any point in the seed model, or in any ModelMeasure, or in any EnergyPlusMeasure, or in a previous ReportingMeasure.

Say you create a measure A with this object above. Clearly you care about the MessagePack only (the json you don't care about, and the cbor I don't know why you turn it off, probably because having CBOR + MessagePack is generally pointless)

Say I (another independent developer) creates a measure B that actually requests specifically for the CBOR output, since I actually make use of it.

Now assume a user creates a workflow, and puts measure B then measure A. Now measure B doesn't get its wanted CBOR output and will throw.

That's why I'm saying we should accommodate for the greater set of requirements whenever possible (and warn if you can't). For OutputJSON (and some other ones) that is perfectly feasible since it's an object with just opt-in flags.

Reply:
I'm ok if people say it's a waste of time to catch these and just hope for the best. But it shouldn't take long to implement the merge functions (and on the C++ side I would probably do it, especially since I have FieldEnums available... here I guess I don't care too much)

Reply:
> I find it particularly confusing to use the ||= operator. If CBOR was previously set to true and then you try to set it to false in your reporting measure, it will remain true... how is that a good or expected outcome?
> 
> ```
> irb(main):006:0> foo = true
> irb(main):007:0> foo ||= false
> irb(main):008:0> foo
> => true
> ```

I had missed this comment. I don't see why a reporting measure would be able to break any other reporting measures or workflows. If you really wanted to completely disable CBOR since don't use a reporting measure that enables it. And change it in the seed model or use a ModelMeasure that switches it off?

Reply:
We'll have to agree to disagree. I often want to disable outputs for reasons of performance or disk space. That is the whole reason that the [OutputControl:Files object was created](https://github.com/NREL/EnergyPlus/pull/7904), for example.